### PR TITLE
docs(link): rfc-002

### PR DESCRIPTION
**Pull request by alhayward**
## Description
* Request for Change (RFC) for Possible Match with Log Odds Score Threshold Range

## Related Issues
#139 

## Additional Notes
#18 

The language around the specific metric calculated from the distribution of Log Odds Scores (e.g., median) is intentionally broad, in case the Data Scientist on the DIBBs 4 Team would like to further experiment with other metrics beyond median.

<--------------------- REMOVE THE LINES BELOW BEFORE MERGING --------------------->

## Checklist
Please review and complete the following checklist before submitting your pull request:

- [ ] I have ensured that the pull request is of a manageable size, allowing it to be reviewed within a single session.
- [ ] I have reviewed my changes to ensure they are clear, concise, and well-documented.
- [ ] I have updated the documentation, if applicable.
- [ ] I have added or updated test cases to cover my changes, if applicable.
- [ ] I have minimized the number of reviewers to include only those essential for the review.

## Checklist for Reviewers
Please review and complete the following checklist during the review process:

- [ ] The code follows best practices and conventions.
- [ ] The changes implement the desired functionality or fix the reported issue.
- [ ] The tests cover the new changes and pass successfully.
- [ ] Any potential edge cases or error scenarios have been considered.

### CODE REVIEW

File: rfc/002-log-odds-score-threshold.md [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+
+* **Explainability.** While using a metric of Log Odds Score may be more understandble among certain users such as public health epidimiologists, this approach to determine Cluster Membership may be less explainable than the Belongingness Ratio (e.g., percentage of existing Patient records in the Person Cluster with which the incoming Patient record matches). However, if a STLT would like to use a more explainable approach to Record Linkage, they have the option of using the DIBBs Basic (deterministic) Algorithm to determine matches, which does not use Log Odds Scores.
+
+* **DIBBs Basic Algorithm.** If a STLT does not wish to use functions with Log Odds Scores, i.e. use the DIBBs Basic Algorithm instead, then a Match Rule function that is not evaluating the sum of Log Odds Scores against a cutoff (e.g., `recordlinker.linking.matchers.rule_probabilistic_match`) will be used (e.g., `recordlinker.linking.matchers.rule_match`). In that case, the same behavior from the Feature Comparison and Match Rule functions of the deterministic DIBBs Basic Algorithm applies. See **Caveats** in [#18](https://github.com/CDCgov/RecordLinker/issues/18) for more information.
```
Comment (ericbuckley) [inline rfc/002-log-odds-score-threshold.md:90]: I'm not sure I follow this, are you suggesting we propoate the summation of feature matches and use that in our latter comparisions when a non-probablistic algorithm is configured OR are you suggesting that we remove the ability to match patient records using non-probablistic algorithms?

File: rfc/002-log-odds-score-threshold.md [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+* Implementing Possible Match (and thus Manual Review) at a *pass level* can create additional manual work. This manual work may also block the Record Linkage algorithm from determining a match decision, which can create bottlenecks for required Manual Reviews as the MPI grows. To arrive at a match decision, the user may need to reconcile many Possible Matches between different permutations of record pairs.
```
Comment (ericbuckley) [inline rfc/002-log-odds-score-threshold.md:81]: I'm a bit unclear as to what the proposal is here for passes.  Are you proposing we only have 1 evaluation pass per algorithm going forward?

File: rfc/002-log-odds-score-threshold.md (hunk 1 above)
Comment (alhayward) [reply rfc/002-log-odds-score-threshold.md:90]: Not quite. When a non-probabilistic algorithm is chosen (a.k.a. the DIBBs Basic Algorithm is used), I think we should not make any changes. That is, we should keep the current implementation of Possible Match for the DIBBs Basic Algorithm (Short-Term Approach with Belongingness Ratio), or any algorithms that do not use log odds scores in the feature comparison functions. This is because the Log Odds Score Threshold Range approach proposed in this RFC cannot apply when log odds scores are not used in the algorithm.

Comment (ericbuckley) [reply rfc/002-log-odds-score-threshold.md:90]: So the resulting payload would change (sometimes belongingness ratio and other times log-odds distance) depending on the configuration of the Algorithm.  Speciffically the log-odds distance value would only ever be returned if all of the rule functions, in all of the passes, use a probablistic match function?

Comment (alhayward) [reply rfc/002-log-odds-score-threshold.md:90]: This sounds right. If any probabilistic functions are used, log odds scores would be evaluated/returned as the "score". If no probabilistic functions are used, Belongingness Ratios would be evaluated/returned as the "score". The resulting payload can still return a general value called "score", but the value would be different depending on whether the user chose a deterministic or probabilistic algorithm.

In this way, we could return a generic "score" in the payload, regardless of whether a deterministic or probabilistic algorithm is used. Just that when the former is used, that score is the Belongingness Ratio (and should be interpreted as such), and when the latter is used, that score is the (median) Log Odds Score. In this approach, in the algorithm configuration, we could generalize the language to have users set a "score threshold" (which under the hood is either a Belongingness Ratio threshold range, or a Log Odds Score threshold range, depending on whether they chose a deterministic or probabilistic algorithm, respectively).

I think the case where the user wants to use a mix of deterministic and probabilistic match functions (e.g., deterministic in Pass 1 and probabilistic in Pass 2) is less likely and should be less supported because it's less explainable. There would be two concepts of "belongingness" (Pass 1 would use Belongingness Ratio, whereas Pass 2 would use Log Odds Score). Also, I recall we pulled Belongingness Ratio out to the algorithm configuration-level (rather than the pass-level), which would support this choice.

Comment (ericbuckley) [reply rfc/002-log-odds-score-threshold.md:90]: Thanks @alhayward.  I've been thinking a lot about the score concept and believe some generic score value (different from belongingness ratio and/or log odds score) would benefit our users.  Particularly with how this could align with the concept of [similarity scores in FHIRs patient matching specification](https://www.hl7.org/fhir/us/identity-matching/patient-matching.html#scoring-matches--responders-system-match-output-quality-score).

File: rfc/002-log-odds-score-threshold.md [hunk 3]
```diff
@@ -0,0 +1,115 @@
+# RFC: Possible Match with Log Odds Score Threshold Range
+
+- **Status**: Under Review
```
Comment (ericbuckley) [inline rfc/002-log-odds-score-threshold.md:3]:
```suggestion
- **Status**: Deferred
```

@alhayward I'm good with this proposal as is, thanks for taking the time to write it.  However, I'm not quite sure it's ready for implementation.  I'm a little wary of a scoring system that changes based on the user business rules set in place, I think this could be confusing to our users, and would like to revisit this in the future to adjust or reaffirm that it won't cause confusion.  For the time being, I think we can merge as long as we set the status appropriately.

Comment (alhayward) [reply rfc/002-log-odds-score-threshold.md:3]: Sounds good. I think that makes sense!

I'd also be curious about Brandon's thoughts on this proposal on the DIBBs 4 Team. In one of our previous DIBBs <> IDWA syncs on Record Linkage, he had a few other ideas on what this \*metric\* calculated from the Log Odds Score distribution could be (beyond median).

**Review by ericbuckley — APPROVED** [review]
